homeassistant/components/binary_sensor/zwave.py

```diff
@@ -94,7 +94,8 @@ def should_poll(self):
 
     def value_changed(self, value):
         """Called when a value has changed on the network."""
-        if self._value.value_id == value.value_id:
+        if self._value.value_id == value.value_id or \
```

Comment:
For some reason value.value_id and value.node is not in sync. This was the root of cause for the thermostat and hvac component not updating the frontend.  The value_Id updated when a manual action on the device performed, and value.node when a frontend action for the device where performed.